src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -3061,6 +3074,197 @@ public ConstructorEdit(INamedTypeSymbol oldType)
             }
         }
 
+        private void AnalyzeCustomAttributes(ISymbol? oldSymbol, ISymbol newSymbol, EditAndContinueCapabilities capabilities, ArrayBuilder<RudeEditDiagnostic> diagnostics, ArrayBuilder<SemanticEditInfo>? semanticEdits, Func<SyntaxNode, SyntaxNode?>? syntaxMap, CancellationToken cancellationToken)
+        {
+            var needsEdit = false;
+
+            if (newSymbol is IMethodSymbol newMethod)
+            {
+                if (oldSymbol is not IMethodSymbol oldMethod)
+                {
+                    return;
+                }
+
+                needsEdit |= HasCustomAttributeChanges(oldMethod.GetReturnTypeAttributes(), newMethod.GetReturnTypeAttributes(), newMethod, capabilities, diagnostics);
+            }
+            else if (newSymbol is INamedTypeSymbol { DelegateInvokeMethod: not null } newType)
+            {
+                var oldType = oldSymbol as INamedTypeSymbol;
+                // If this is a delegate with attributes on its return type for example, they are found on the DelegateInvokeMethod
+                AnalyzeCustomAttributes(oldType?.DelegateInvokeMethod, newType.DelegateInvokeMethod, capabilities, diagnostics, semanticEdits, syntaxMap, cancellationToken);
+            }
+
+            foreach (var parameter in newSymbol.GetParameters())
+            {
+                var oldParameter = oldSymbol?.GetParameters().SingleOrDefault(p => p.Name.Equals(parameter.Name));
```

Comment:
>SingleOrDefault [](http://example.com/codeflow?start=62&length=15)

Is it possible there are parameters with duplicate names in error scenarios? We should not crash; just ignore such case.